proposed/http-client/http-client.md

```diff
@@ -126,7 +127,9 @@ use Psr\Http\Message\RequestInterface;
 use Psr\Http\Client\Exception;
 
 /**
- * Thrown when a response was received but the request itself failed.
+ * Thrown when a response was received but the request itself failed. This exception MAY
+ * be thrown on HTTP response codes 4xx and 5xx. This exception MUST NOT be thrown when 
```

Comment:
not sure what CS fig uses, but imho its best to have a single line with the general info, then a blank line and then more information.